### ISSUE TITLE
Move INotificationsClient under IActivitiesClient

### ISSUE THREAD

**Issue by shiftkey** [octokit/octokit.net#1015]
Currently it's a top-level client:

```csharp
...
var result = await client.Notifications.GetAllForCurrent();
```

But the [API documentation](https://developer.github.com/v3/activity/notifications/) has it located under `Activities`:

<img width="363" alt="screen shot 2015-12-18 at 11 17 10 am" src="https://cloud.githubusercontent.com/assets/359239/11886357/38144112-a579-11e5-93a7-0e10bb7c944e.png">

We should do the same - but mark the top-level property as `[Obsolete]` for the next release to help with transitioning.

```csharp
var result = await client.Activity.Notifications.GetAllForCurrent();
```

**Comment by Haacked** [comment]
:+1:

**Comment by M-Zuber** [comment]
I'd like to try and take this

**Comment by shiftkey** [comment]
Fixed in #1019
